### PR TITLE
Dockerfile.arm64 Catch up Dockerfile.amd64

### PR DESCRIPTION
**Pull request by LyleLee**
Adopt the same iptable version

With CentOS8 or latest debian OS, Whose iptables implementation
have change from Xtables to nftables. Need to pack legacy
iptables into node image.

## Description

+ Calico/node aarh64 image iptables version is different from amd64 images.
+ Iptables rules not writing into host.
+ Ping between pod not working.

Try to fix issues:
https://github.com/projectcalico/node/issues/524

## Todos
- [x] Tests
- [x] Documentation
- [x] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -53,8 +54,8 @@ FELIX_GPL_SOURCE=filesystem/included-source/felix-ebpf-gpl.tar.gz
 INCLUDED_SOURCE=$(BIRD_SOURCE) $(FELIX_GPL_SOURCE)
 
 # Versions and locations of dependencies used in tests.
-CALICOCTL_VER?=master
-CNI_VER?=master
+CALICOCTL_VER?=v3.14.1
```
Comment (caseydavenport) [inline Makefile:57]: These should be `master` still, since this is the Makefile for the master branch.

File: Makefile [hunk 2]
```diff
@@ -18,7 +18,7 @@ Makefile.common: Makefile.common.$(MAKE_BRANCH)
 Makefile.common.$(MAKE_BRANCH):
 	# Clean up any files downloaded from other branches so they don't accumulate.
 	rm -f Makefile.common.*
-	curl --fail $(MAKE_REPO)/Makefile.common -o "$@"
+	curl -k --fail $(MAKE_REPO)/Makefile.common -o "$@"
```
Comment (caseydavenport) [inline Makefile:21]: What is this change for?

File: Makefile [hunk 3]
```diff
@@ -1,5 +1,5 @@
 PACKAGE_NAME?=github.com/projectcalico/node
-GO_BUILD_VER?=v0.45
+GO_BUILD_VER?=v0.40
```
Comment (caseydavenport) [inline Makefile:2]: I don't think we should revert the makefile version. Why did you need to do this?

File: Makefile [hunk 4]
```diff
@@ -164,6 +165,8 @@ remote-deps: mod-download
 		cp -r `go list -mod=mod -m -f "{{.Dir}}" github.com/projectcalico/felix`/bpf-apache bin/bpf; \
 		chmod -R +w bin/bpf; \
 		chmod +x bin/bpf/bpf-gpl/list-* bin/bpf/bpf-gpl/calculate-*; \
+		sed -i "s/amd64/arm64/" ./bin/bpf/bpf-gpl/Makefile; \
```
Comment (caseydavenport) [inline Makefile:168]: Hm, I'm not sure if this is a reliable way to go about this.

I think we'd want to enhance that Makefile to have support for multi-arch in a separate PR. It lives here: https://github.com/projectcalico/felix/blob/master/bpf-gpl/Makefile

**Review by caseydavenport — CHANGES_REQUESTED** [review]
Added a few comments. There are some changes that I think aren't related to this PR, some commented lines of code, etc., that I don't fully understand.

File: Makefile (hunk 1 above)
Comment (LyleLee) [reply Makefile:57]: Agree, the specific v3.14.1 thing is just the tag my special case needs when the latest stable is v3.14.1. A universal match to `master` should be our goal.

File: Makefile (hunk 2 above)
Comment (LyleLee) [reply Makefile:21]: My internet needs the proxy to go out through, and  `-k` helps me get rip of errors pop out every time I run make.  That was my bad not removing it in the patch.

File: Makefile (hunk 3 above)
Comment (LyleLee) [reply Makefile:2]: Well, this is just my cook to make it work. Thanks to point out.

File: Makefile (hunk 4 above)
Comment (LyleLee) [reply Makefile:168]: Totally agree.

File: Dockerfile.arm64 [hunk 5]
```diff
@@ -11,39 +11,138 @@
...
 
 # Enable non-native builds of this image on an amd64 hosts.
 # This must be the first RUN command in this file!
 # we only need this for the intermediate "base" image, so we can run all the apk and other commands
 # when running on a kernel >= 4.8, this will become less relevant
-COPY --from=qemu /usr/bin/qemu-aarch64-static /usr/bin/
-
-# Install remaining runtime deps required for felix from the global repository
-RUN apk add --no-cache ip6tables ipset iputils iproute2 conntrack-tools runit ca-certificates
+#COPY --from=qemu /usr/bin/qemu-aarch64-static /usr/bin/
```
[the remark this answers is not in the document]
Comment (LyleLee) [reply Dockerfile.arm64:130]: No, it is not my intent.  it is something left behind because of my rash.

**Comment by LyleLee** [comment]
> Added a few comments. There are some changes that I think aren't related to this PR, some commented lines of code, etc., that I don't fully understand.

I will do my best to submit a new PR, though I do doubt about my ability in this area.  That will be nice if the developer could make this change happen.

**Comment by caseydavenport** [comment]
This PR is a bit out of date now, so I'm going to close it due to inactivity. 

Would be happy to take another look at this if it gets updated.

**Comment by lwr20** [comment]
Likely covered by #1044
